### PR TITLE
Resolves 12. As a user, I want to view a list of my shopping list items in order of how soon I am likely to need to buy each of them again so that it’s clear what I need to buy soon.

### LINKED ISSUE #12: 12. As a user, I want to view a list of my shopping list items in order of how soon I am likely to need to buy each of them again so that it’s clear what I need to buy soon.
Possible item states are as follows:

* Need to buy soon (fewer than 7 days)
* Need to buy kind of soon (between 7 & 30 days)
* Need to buy not soon (more than 30 days)
* Inactive (when there’s only 1 purchase in the database or the purchase is really out of date [the time that has elapsed since the last purchase is 2x what was estimated])

AC:

* Items in the list are shown as visually distinct (e.g., with a different background color on the list item) according to how soon the item is expected to be bought again: Soon, Kind of soon, Not soon, Inactive
* Items should be sorted by the estimated number of days until next purchase
* Items with the same number of estimated days until next purchase should be sorted alphabetically
* Items in the different states should be described distinctly when read by a screen reader

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description

This PR will organize our list by time to next purchase and then alphabetically by item within each purchase. Additonally it will color code red, yellow, green, or grey depending on the item's estimated purchase range per the desired AC numbers. In this PR we had to learn a lot about queries with firestore, specifically compound queries with more than one index. In the end however we decided to create a client side function with the built in sort method to evaluate purchase estimates and compare to the list array containing our database. We chose this route so that we could focus on the color coding locally and in realtime while reducing the I/O with the Firestore database. This allowed for an easy color coding method that will update each time the user looks at the list. We may want to refactor this into a switch case should our styling get more complex or our color status get more specific,

Additionally we added some basic accessibility with aria-labels and also made a few changes to other UI elements and labels to be more accessible per MDN docs here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG
  

## Related Issue

closes #12 

## Acceptance Criteria

* Items in the list are shown as visually distinct (e.g., with a different background color on the list item) according to how soon the item is expected to be bought again: Soon, Kind of soon, Not soon, Inactive
* Items should be sorted by the estimated number of days until next purchase
* Items with the same number of estimated days until next purchase should be sorted alphabetically
* Items in the different states should be described distinctly when read by a screen reader

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓  | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

filter list

### After

after-shot

## Testing Steps / QA Criteria

* pull the latest commit of this branch into your local IDE
* run the code on a local server in the browser
* try adding a few new items to your list testing for different estimates and alphabetical order per color
* on a mac press 'cmd + F5' to activate a screen reader, for windows download NVDA screen reader here: https://www.nvaccess.org/download/ and follow instructions - once the screen reader is active mouse or tab around to hear labeled elements
* use chrome or firefox dev tools to inspect accessibility properties and check for suggestions

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -20,3 +20,19 @@
...
+  background-color: yellow;
+}
+
+.not-soon {
+  background-color: red;
+}
+
+.inactive {
+  background-color: lightgrey;
+}
```
Comment (MelinaMejia95) [inline src/App.css:38]: I love this ❤️

**Review by MelinaMejia95 — APPROVED** [review]
I love the comments you added! It's really helpful when you're doing code review. Such an amazing work team ❤️

File: src/components/ShoppingList/PopulatedList.js [hunk 2]
```diff
@@ -16,11 +20,34 @@ const PopulatedList = () => {
...
+      // Sort by days until next purchase
+      // The if/else ensures that 'null' (inactive) items will always be at bottom
+      list.sort(function (a, b) {
+        if (a.purchaseCountdown === null) {
+          return 1;
+        } else if (b.purchaseCountdown === null) {
+          return -1;
+        }
+        return a.purchaseCountdown < b.purchaseCountdown ? -1 : 1;
+      });
```
Comment (Maeesha-Rahman) [inline src/components/ShoppingList/PopulatedList.js:48]: nice use of `sort()`!

**Review by Maeesha-Rahman — APPROVED** [review]
Awesome work! I love the comments, made things easier to understand. Also nice job on going through the codebase and adding aria-labels to inputs that didn't already have one. This will definitely improve the accessibility of the app!

**Review by wlcreate — APPROVED** [review]
Awesome job! I love the variable naming going on and comment to make it easier to understand what each piece is doing 😍 Also, I appreciate in your PR your reasoning and explanation behind creating a client side function with sort.

**Pull request merged**